core/translation.cpp

```diff
@@ -839,9 +844,17 @@ void Translation::_set_messages(const DVector<String>& p_messages){
 
 void Translation::set_locale(const String& p_locale) {
 
-	ERR_EXPLAIN("Invalid Locale: "+p_locale);
-	ERR_FAIL_COND(!is_valid_locale(p_locale));
-	locale=p_locale;
+	if(!is_valid_locale(p_locale)) {
```

Comment:
As in falling back to a default language like English if the locale string is unrecognized and the language it specifies is unrecognized?